### ISSUE TITLE
Delete objects (purge) function leaves inconsistencies in target project metadata

### ISSUE THREAD

**Issue by grololo06** [oceanomics/ecotaxa_dev#393]
I've noticed, that, after a DELETEALL in a test project, leaving the project for reuse:
- loadedfiles still has a value
- all mappings are still set to previous imports

This can be seen in Admin screen, full view for the project.
As a result, you basically cannot reuse the project for many other imports ,as you might hit the mapping limit, and previous recorded loadedfiles makes re-import a bit tricky.

**Comment by jiho** [comment]
This is kind of expected. Deleting all objects from a project should delete the objects but leave the metadata intact (for future re-import for example) If a brand new project is needed, it can just be created, can't it?

**Comment by grololo06** [comment]
Use case:
- Import a sample with custom columns sample_ls01, sample_ls02,.... to sample_ls20

Purge all
- Try re-import same sample but custom columns are now sample_jo01, sample_jo02... sample_jo20

Import code will assume the need for 40 sample custom columns thus fail as we have only 30.
I think that mappings should map something, so no data means no mapping. Problem is that some mappings are also referenced in project preferences.

**Comment by grololo06** [comment]
The problem gets even worse if one deletes only some objects inside a project, as e.g. "loaded files" becomes partially wrong but we've lost the link files<->objects so no way to fix this data.

**Comment by jiho** [comment]
Proposed ways to fix the first issue:
- force the deletion of a project after DELETEALL
- limit "Delete" to deleting *some* objects and make "Delete project" the only option to delete all objects+the project.
I think this is the most realistic use case for deleting all objects in a project, outside of tests, and it solves the "remaining mapping and other cruft" problem.

**Comment by jiho** [comment]
As for the lost link between imported files and deletion of objects, the only use case I see is
1. I import several files in batch from a folder
2. I delete some objects that happen to correspond exactly to the content of one file (i.e. a sample and each file is a sample)
3. I realise this was a mistake and want to reimport only the file containing the objects I deleted but I don't know which one it is exactly so I select the whole folder and tick "Skip already imported tsv files"

This is a edge case scenario, would require to track which file each object comes from and delete the record of the file being imported only when all objects have been deleted. Plus, in that case, there is a slower but more powerful (and working solution): reimport with "Skip objects that have already been imported".

So basically I think we don't care.

**Comment by picheral** [comment]
The option is "Skip **objects** that have already been imported " which is efficient for this case.
I also agree on the DELETE / DELETE Project separate procedure.

**Comment by jiho** [comment]
To finalise a specification:

1. In the Filtered menu, "Delete objects" leads to the current page, on which the DELETEALL and Delete project after DELETEALL have been removed.

2. In the Project menu, "Delete objects or project" disappears. At the bottom of Project settings, a section with

**Delete project**

To delete this project, type its name below and click Delete

[ Text box in which the user must type/copy-paste the name of the project]

<kbd>Delete</kdb>

i.e. model the deletion of a project after the deletion of a repository on GitHub.

**Comment by grololo06** [comment]
Will be fixed by implementing #108 -> duplicate
